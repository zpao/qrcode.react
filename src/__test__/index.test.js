import React from 'react';
import renderer from 'react-test-renderer';
const QRCode = require('qrcode.react');

it('renders correctly', () => {
  const tree = renderer
    .create(
      <QRCode
        value={'http://picturesofpeoplescanningqrcodes.tumblr.com/'}
        size={128}
        bgColor={'#ffffff'}
        fgColor={'#000000'}
        level={'L'}
        includeMargin={false}
        renderAs={'svg'}
        imageSettings={{
          src: 'https://static.zpao.com/favicon.png',
          x: null,
          y: null,
          height: 24,
          width: 24,
          excavate: true,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
