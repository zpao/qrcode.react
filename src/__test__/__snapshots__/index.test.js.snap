// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<svg
  height={128}
  shapeRendering="crispEdges"
  viewBox="0 0 29 29"
  width={128}
>
  <path
    d="M0,0 h29v29H0z"
    fill="#ffffff"
  />
  <path
    d="M0 0h7v1H0zM8 0h1v1H8zM10 0h5v1H10zM17 0h2v1H17zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM9 1h2v1H9zM13 1h1v1H13zM19 1h2v1H19zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM10 2h2v1H10zM13 2h3v1H13zM19 2h2v1H19zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM11 3h3v1H11zM16 3h2v1H16zM19 3h2v1H19zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h1v1H9zM12 4h1v1H12zM16 4h2v1H16zM19 4h1v1H19zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM10 5h2v1H10zM13 5h1v1H13zM18 5h1v1H18zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM8 7h1v1H8zM10 7h1v1H10zM13 7h1v1H13zM15 7h2v1H15zM19 7h2v1H19zM0 8h2v1H0zM3 8h2v1H3zM6 8h1v1H6zM9 8h4v1H9zM16 8h2v1H16zM19 8h2v1H19zM22 8h1v1H22zM28,8 h1v1H28zM0 9h1v1H0zM4 9h2v1H4zM8 9h2v1H8zM11 9h1v1H11zM13 9h3v1H13zM18 9h2v1H18zM21 9h1v1H21zM23 9h2v1H23zM26 9h2v1H26zM0 10h4v1H0zM6 10h1v1H6zM8 10h1v1H8zM10 10h2v1H10zM13 10h1v1H13zM15 10h2v1H15zM18 10h1v1H18zM21 10h1v1H21zM24 10h1v1H24zM26 10h1v1H26zM0 11h1v1H0zM2 11h1v1H2zM5 11h1v1H5zM8 11h3v1H8zM19 11h2v1H19zM24 11h1v1H24zM28,11 h1v1H28zM1 12h3v1H1zM5 12h2v1H5zM9 12h1v1H9zM20 12h1v1H20zM22 12h1v1H22zM28,12 h1v1H28zM1 13h1v1H1zM3 13h1v1H3zM8 13h1v1H8zM18 13h1v1H18zM22 13h4v1H22zM27,13 h2v1H27zM0 14h2v1H0zM3 14h2v1H3zM6 14h2v1H6zM18 14h3v1H18zM22 14h1v1H22zM24 14h3v1H24zM28,14 h1v1H28zM1 15h1v1H1zM3 15h1v1H3zM8 15h2v1H8zM21 15h4v1H21zM26 15h1v1H26zM28,15 h1v1H28zM2 16h1v1H2zM6 16h1v1H6zM8 16h1v1H8zM10 16h1v1H10zM20 16h1v1H20zM0 17h1v1H0zM5 17h1v1H5zM9 17h1v1H9zM18 17h4v1H18zM24 17h4v1H24zM0 18h2v1H0zM3 18h5v1H3zM11 18h3v1H11zM15 18h1v1H15zM19 18h4v1H19zM28,18 h1v1H28zM0 19h2v1H0zM3 19h3v1H3zM10 19h2v1H10zM13 19h1v1H13zM18 19h5v1H18zM24 19h3v1H24zM0 20h8v1H0zM9 20h1v1H9zM11 20h2v1H11zM14 20h2v1H14zM17 20h1v1H17zM19 20h9v1H19zM8 21h1v1H8zM11 21h5v1H11zM17 21h4v1H17zM24 21h2v1H24zM0 22h7v1H0zM9 22h1v1H9zM11 22h2v1H11zM15 22h2v1H15zM19 22h2v1H19zM22 22h1v1H22zM24 22h2v1H24zM0 23h1v1H0zM6 23h1v1H6zM10 23h1v1H10zM12 23h1v1H12zM14 23h2v1H14zM17 23h4v1H17zM24 23h2v1H24zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM8 24h4v1H8zM15 24h2v1H15zM19 24h6v1H19zM27,24 h2v1H27zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM8 25h4v1H8zM13 25h7v1H13zM21 25h1v1H21zM25,25 h4v1H25zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM10 26h1v1H10zM12 26h1v1H12zM15 26h2v1H15zM18 26h1v1H18zM21 26h1v1H21zM23 26h3v1H23zM27,26 h2v1H27zM0 27h1v1H0zM6 27h1v1H6zM8 27h1v1H8zM14 27h2v1H14zM21 27h1v1H21zM23 27h4v1H23zM28,27 h1v1H28zM0 28h7v1H0zM8 28h3v1H8zM12 28h2v1H12zM19 28h2v1H19zM22 28h2v1H22z"
    fill="#000000"
  />
  <image
    height={5.4375}
    preserveAspectRatio="none"
    width={5.4375}
    x={11.78125}
    xlinkHref="https://static.zpao.com/favicon.png"
    y={11.78125}
  />
</svg>
`;
